### ISSUE TITLE
Email settings

### ISSUE THREAD

**Issue by gondzo** [appirio-tech/connect-app#1991]
Enable email settings in notification preferences and add bundling options. This is dependent on changes in https://github.com/topcoder-platform/tc-notifications/issues/40

**Comment by maxceem** [comment]
#### TL;DR
Current shape of data which we get/send via API doesn't have a good place to put settings for email bundling. As we have only places for settings per notification types, but don't have place for common settings. The question is here, whether we have to change shape of data passing by API and break backward compatibilty, or try to add settings for email bundle somewhere even it breakes semantic meaning of passing objects and duplicates data.

#### Details
Currently, API calls to get notification settings returns an object with notification types as keys:
   ```
    {
       "notifications.connect.project.updated:{"email":"yes","web":"no"},
       "notifications.connect.project.created":{"email":"yes","web":"no"},
       ...
    }
   ```
   and call to save notification settings sends an array of settings per each notification type and delivery type:
   ```js
    [{
      "topic": "notifications.connect.project.created",
      "deliveryMethod": "email",
      "value": "yes"
    }, {
      "topic": "notifications.connect.project.created",
      "deliveryMethod": "web",
      "value": "no"
    }, {
      "topic": "notifications.connect.project.updated",
      "deliveryMethod": "email",
      "value": "yes"
    }, {
      "topic": "notifications.connect.project.updated",
      "deliveryMethod": "web",
      "value": "no"
    }
    ...
   ]
   ```

   Both these ways don't give us a place to send some common settings for all notifications types.

There could be few solutions:
1. Hacky. Add bundle settings for each notification type in API, even though it's common for all notification types:
    ```
    // get
    {
       "notifications.connect.project.updated:{"email":"yes","web":"no","emailBundle":"24h"},
       "notifications.connect.project.created":{"email":"yes","web":"no","emailBundle":"24h"},
       ...
    }
    // send
    [{
      "topic": "notifications.connect.project.created",
      "deliveryMethod": "email",
      "value": "yes",
      "bundle: "24h"
    }, {
      "topic": "notifications.connect.project.created",
      "deliveryMethod": "web",
      "value": "no"
    }, 
   ```
    - pro:
       - backward compatibility of API
       - potentially we support bundle settings per notification type
    - cons:
       - in this object `{"email":"yes","web":"no","bundleEmail":"24h"},` properties `email` and `web` are semanticaly different from `emailBundle` and in the code we iterate by them, we have to exclude `emailBundle` when iterating. So it's kind of hacky solution
       - we duplicate same data when sending
 2. Or brake backward compatibility of API and change the shape of data:
     ```
     // get
     {
        "notificationTypes": {
          "notifications.connect.project.updated:{"email":"yes","web":"no"},
          "notifications.connect.project.created":{"email":"yes","web":"no"},
          ...
        },
       "bundleEmail": "24h"
     }
     ```
     or more general to keep in mind future services like Slack and so on
     ```
     // get
     {
        "notificationTypes": {
          "notifications.connect.project.updated:{"email":"yes","web":"no"},
          "notifications.connect.project.created":{"email":"yes","web":"no"},
          ...
        },
       "services": {
          "email": {
            "bundle": "24h"
          },
          "slack": {
            ...
          }
       }
     }
     ```
    or even more general adding support to config more settings per notification type in future:
     ```
     // get
     {
        "notificationTypes": {
          "notifications.connect.project.updated: {
              "services": {
                 "email": {
                    "enabled": true
                    ... // other settings for email service for particular notification type
                  },
                  "web": {
                    "enabled": true
                  }
              }   
          }          
          ...
        },
       "services": {
          "email": {
            "bundle": "24h"
            ... // other common settings for email service
          },
          "slack": {
            ...
          }
       }
     }
     ```

     And for sending minimal change could be like this:
     ```
     // send
     {
       notificationTypes: [{
          "topic": "notifications.connect.project.created",
          "deliveryMethod": "email",
          "value": "yes"
        }, {
         "topic": "notifications.connect.project.created",
         "deliveryMethod": "web",
         "value": "no"
       }, 
       "bundleEmail": "24h"
     }
    ```

     Also, I don't see a reason why shape of get/send data is different, so maybe in makes sense to unify them if we already changing API and choose one of the shapes for both sending and getting data.

**Comment by gondzo** [comment]
I think we can break the api compatibility now as we use these settings only in connect andgo with the most robust option (last one in your example). It will require some db changes, so can you just create a migration script for new/altered tables

**Comment by maxceem** [comment]
@gondzo In `tc-notifications` project [migartions folder](https://github.com/topcoder-platform/tc-notifications/tree/dev/migrations) contains `sql` scripts instead of seqelize migration scripts. Also, there is no `sequelize-cli` in dependaecy list. So it looks like migrations were run manually or something.

Should I create simillar `sql` script or add `sequelize-cli` as dependancy and create `sequelize` migration script?

Also how migrations are going to be run? If using `sequelize-cli`, probably `node_modules/.bin/sequelize db:migrate` has to be added to `deploy.sh` (work only on AWS) or `npm start` script (also will work locally).

**Comment by gondzo** [comment]
Db changes are done manually for now, so we just need a new .sql script, not sequelize-cli migration

**Comment by vikasrohit** [comment]
@gondzo can we take this opportunity to use proper sequelize migration scripts for `tc-notifications`?

**Comment by maxceem** [comment]
FYI for small screens I adopt it like this:

![image](https://user-images.githubusercontent.com/146016/39955557-7c97180e-5603-11e8-9233-08401e3dbfdd.png)

**Comment by vikasrohit** [comment]
Looks good to me. Thanks @maxceem  for the improvisation.
